### PR TITLE
Stun is now a status effect

### PR DESCRIPTION
**Pull request by ChangelingRain**
2/4 Complete

### CODE REVIEW

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -298,6 +298,8 @@ Proc for attack log creation, because really why not
 	var/endtime = world.time + delay
 	var/starttime = world.time
 	. = 1
+	var/mob/living/L = user
+	var/is_living = istype(L)
```
Comment (lzimann) [inline code/__HELPERS/mobs.dm]: Why not just define IsStun() on /mob returning `TRUE` and override on living? Would avoid doing this.

File: code/_onclick/hud/screen_objects.dm [hunk 2]
```diff
@@ -348,7 +348,8 @@
 /obj/screen/storage/Click(location, control, params)
 	if(world.time <= usr.next_move)
 		return 1
-	if(usr.stat || usr.unconscious || usr.stun || usr.knockdown)
+	var/mob/living/L = usr
```
Comment (lzimann) [inline code/_onclick/hud/screen_objects.dm]: Same here.

File: code/__HELPERS/mobs.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/__HELPERS/mobs.dm]: The default would be FALSE and I'd rather not even pretend that non-living mobs can have stuns.

Comment (lzimann) [inline code/__HELPERS/mobs.dm]: It's not about pretending non-living can have stuns, it's avoiding a typecheck and typecasting that can actually be avoided without any issue.

**Comment by nicbn** [comment]
Add WIP tag.

**Comment by ChangelingRain** [comment]
It's part of a larger project and this part is complete!

File: code/datums/status_effects/debuffs.dm [hunk 3]
```diff
@@ -16,6 +16,10 @@
 	if(update_canmove)
 		owner.update_canmove()
 
+//STUN
+/datum/status_effect/incapacitating/stun
+	id = "stun"
```
Comment (ChangelingRain) [inline code/datums/status_effects/debuffs.dm]: wow it's like two fucking lines to make a status effect now, all you need to do is add checks where you need them, isn't that wonderful

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 4]
```diff
@@ -105,9 +105,7 @@
 
 /mob/living/carbon/human/proc/check_block()
 	if(mind)
-		if(mind.martial_art && mind.martial_art.block_chance \
-		&& prob(mind.martial_art.block_chance) && in_throw_mode \
-		&& !stat && !knockdown && !stun)
+		if(mind.martial_art && mind.martial_art.block_chance && prob(mind.martial_art.block_chance) && in_throw_mode && !stat && !knockdown && !IsStun())
```
Comment (lordpidey) [inline code/modules/mob/living/carbon/human/human_defense.dm]: holy crap, that's a long boolean operation.  Can be slightly shortened though, as the block_chance is made redundant by the prob

`if(mind.martial_art &&  prob(mind.martial_art.block_chance) && in_throw_mode && !stat && !knockdown && !IsStun())`

File: code/__DEFINES/status_effects.dm [hunk 5]
```diff
@@ -30,6 +30,8 @@
 // DEBUFFS //
 /////////////
 
+#define STATUS_EFFECT_STUN /datum/status_effect/incapacitating/stun //the affected is asleep
```
Comment (lordpidey) [inline code/__DEFINES/status_effects.dm]: The comment needs changing.

**Review by lordpidey — CHANGES_REQUESTED** [review]

**Comment by XDTM** [comment]
I wonder if the freeze will prevent rebalancing with this, there's a lot of 2s stuns that could be shorter

File: code/game/machinery/computer/arcade.dm [hunk 6]
```diff
@@ -618,7 +618,9 @@
 				if(emagged) //has to be here because otherwise it doesn't work
 					playsound(loc, 'sound/effects/supermatter.ogg', 100, 1)
 					say("A miniature black hole suddenly appears in front of [src], devouring [usr] alive!")
-					usr.Stun(200) //you can't run :^)
+					if(isliving(usr))
+						var/mob/living/L = usr
+						L.Stun(200, ignore_canstun = TRUE) //you can't run :^)
```
Comment (lzimann) [inline code/game/machinery/computer/arcade.dm]: Was ths `ignore_canstun` change intended(for everything, pretty much).

Comment (ChangelingRain) [inline code/game/machinery/computer/arcade.dm]: Yes. These aren't supposed to be avoidable stuns, especially the suicide ones.

**Review by lzimann — APPROVED** [review]

**Pull request merged**